src/handlers/web3.js

```diff
@@ -40,22 +53,61 @@ export const web3SetHttpProvider = async network => {
   if (network.startsWith('http://')) {
     web3Provider = new JsonRpcProvider(network, NetworkTypes.mainnet);
   } else {
-    web3Provider = new JsonRpcProvider(
-      replace(infuraUrl, 'network', network),
-      network
-    );
+    web3Provider = new JsonRpcProvider(replace(infuraUrl, 'network', network));
+    return web3Provider.ready;
   }
-  return web3Provider.ready;
 };
 
-export const sendRpcCall = async payload =>
-  web3Provider.send(payload.method, payload.params);
+export const isL2Network = network => {
+  switch (network) {
+    case NetworkTypes.arbitrum:
+    case NetworkTypes.optimism:
+    case NetworkTypes.polygon:
+      return true;
+    default:
+      return false;
+  }
+};
 
-export const getTransactionReceipt = txHash =>
-  sendRpcCall({
-    method: 'eth_getTransactionReceipt',
-    params: [txHash],
-  });
+/**
+ * @desc returns a web3 provider for the specified network
+ * @param {String} network
+ */
+export const getProviderForNetwork = async network => {
```

Comment:
note to self: this creates a new provider object every time it is called (and for arbitrum this is called repeatedly for gas prices) - need to recall if this pattern isn't recommended by ethers